compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/MouseWheelScrollable.kt

```diff
@@ -75,12 +79,22 @@ internal class MouseWheelScrollableElement(
 
 internal abstract class MouseWheelScrollNode(
     var scrollingLogic: ScrollingLogic,
-    var mouseWheelScrollConfig: ScrollConfig
+    var mouseWheelScrollConfig: ScrollConfig,
+    private var _enabled: Boolean,
 ) : DelegatingNode(), PointerInputModifierNode {
     private val pointerInputNode = delegate(SuspendingPointerInputModifierNode {
-        mouseWheelInput()
+        if (_enabled) {
+            mouseWheelInput()
+        }
     })
 
+    var enabled
+        get() = _enabled
+        set(value) {
+            _enabled = value
+            pointerInputNode.resetPointerInputHandler()
```

Comment:
Maybe it's better to filter it on events instead of relying on another state?

Reply:
If we have running animations, we should cancel them, restarting the pointerInput coroutine.

This code is the same as in Draggable.